src/matrixrtc/CallMembership.ts

```diff
@@ -120,109 +207,233 @@ const checkSessionsMembershipData = (
+                return this.parentEventData.sender;
+        }
     }
 
-    public get eventId(): string | undefined {
-        return this.parentEvent.getId();
+    public get eventId(): string {
+        return this.parentEventData.eventId;
     }
 
     /**
-     *  @deprecated Use sessionDescription.id instead.
+     * The slot id to find all member building one session `slot_id` (format `{application}#{id}`).
+     * This is computed in case SessionMembershipData is used.
      */
-    public get callId(): string {
-        return this.membershipData.call_id;
+    public get slotId(): string {
+        const { kind, data } = this.membershipData;
+        switch (kind) {
+            case "rtc":
+                return data.slot_id;
+            default: // "session":
+                return slotDescriptionToId({ application: this.application, id: data.call_id });
+        }
     }
 
     public get deviceId(): string {
-        return this.membershipData.device_id;
+        const { kind, data } = this.membershipData;
+        switch (kind) {
+            case "rtc":
+                return data.member.device_id;
+            default: // "session":
+                return data.device_id;
+        }
     }
 
     public get callIntent(): RTCCallIntent | undefined {
-        return this.membershipData["m.call.intent"];
+        return this.membershipData.data["m.call.intent"];
     }
 
-    public get sessionDescription(): SessionDescription {
-        return {
-            application: this.membershipData.application,
-            id: this.membershipData.call_id,
-        };
+    /**
+     * Parsed `slot_id` (format `{application}#{id}`) into its components (application and id).
+     */
+    public get slotDescription(): SlotDescription {
+        return slotIdToDescription(this.slotId);
     }
 
-    public get application(): string | undefined {
-        return this.membershipData.application;
+    public get application(): string {
+        const { kind, data } = this.membershipData;
+        switch (kind) {
+            case "rtc":
+                return data.application.type;
+            default: // "session":
+                return data.application;
+        }
+    }
+    public get applicationData(): { type: string } & Record<string, any> {
+        const { kind, data } = this.membershipData;
+        switch (kind) {
+            case "rtc":
+                return data.application;
+            default: // "session":
+                return { "type": data.application, "m.call.intent": data["m.call.intent"] };
+        }
     }
 
+    /** @deprecated scope is not used and will be removed in future versions. replaced by application specific types.*/
     public get scope(): CallScope | undefined {
-        return this.membershipData.scope;
+        const { kind, data } = this.membershipData;
+        switch (kind) {
+            case "rtc":
+                return undefined;
+            default: // "session":
```

Comment:
If default is meant to also catch session then this looks nicer to my eye:

```suggestion
            case "session":
            default:
```